rplugin/python3/deoplete/source/buffer.py

```diff
@@ -29,8 +29,12 @@ def on_event(self, context):
     def gather_candidates(self, context):
         self.__make_cache(context)
 
-        buffers = [x['candidates'] for x in self.__buffers.values()
-                   if x['filetype'] in context['filetypes']]
+        buffer_values = self.__buffers.values()
+        if context['vars'].get('require_same_filetype', True):
+          buffer_values = [x for x in buffer_values
```

Comment:
Righteous tweaks! :+1:  Go ahead with your PR because this patch doesn't seem to cut it anyway. :8ball: